webpack/assets/javascripts/react_app/components/FactCharts/index.js

```diff
@@ -20,8 +22,10 @@ const ConnectedFactChart = ({ id, path, title, search }) => {
   const modalToDisplay = useSelector(state => selectDisplayModal(state, id));
   const dispatch = useDispatch();
   const dispatchCloseModal = () => dispatch(closeModal(id));
-  const dispatchOpenModal = () =>
-    dispatch(openModal({ id, title, apiKey: key, apiUrl: path }));
+  const dispatchOpenModal = () => {
+    dispatch(get({ key, url: path }));
+    dispatch(openModal({ id, title }));
```

Comment:
Previously, the `openModal` action was confusingly dispatching two other actions. While I think it's possible to also do that with the slice pattern, i find it clearer to dispatch the actions separately.

Reply:
Good